### ISSUE TITLE
feat: iOS: make BeagleScreenViewController -> convenience init(_ component: ServerDrivenComponent..) public

### ISSUE THREAD

**Issue by iSallyAbdallah** [ZupIT/beagle#1831]
## Use case
I need to create BeagleScreenViewController from ServerDrivenComponent , please make it public 

## Proposal
```
  public convenience init(_ component: ServerDrivenComponent, controllerId: String? = nil, config: BeagleConfiguration = GlobalConfiguration) {
        self.init(viewModel: .init(screenType: .declarative(component.toScreen()), resolver: config.resolver), controllerId: controllerId, config: config)
        self.navigationControllerId = controllerId
    }
    
```

**Comment by dantes-git** [comment]
Hi @iSallyAbdallah 

This function is internal because our intention is for the developer to build a server driven solution, in this case, build the components on the backend. You can use this method that load a component from a endpoint:
`BeagleScreenViewController(_ remote: ScreenType.Remote, controllerId: String? = nil)`

Or you can also load from a json encoded in a String:
`BeagleScreenViewController(_ json: String, controllerId: String? = nil)`

I didn't understand in which scenario you need to create the `BeagleScreenViewController` passing the component. Could you exemplify? Perhaps there is a better alternative to your problem.

**Comment by iSallyAbdallah** [comment]
Hi @dantes-git 

I create a custom component  called card Stack and each card loaded should be ServerDrivenComponent, I need to create BeagleScreenViewController for each card due to the widget load array of ServerDrivenComponent

example of JSON : 
```
 {
          "_beagleComponent_": "custom:cardStack",
          "cards": [
            {
              "_beagleComponent_": "beagle:container",
              "style": {
                "size": {
                  "width": {
                    "value": 100,
                    "type": "PERCENT"
                  }
                },
                "backgroundColor": "#888888"
              },
              "children": [
                {
                  "_beagleComponent_": "beagle:image",
                  "path": {
                    "_beagleImagePath_": "remote",
                    "url": "https://s.w-x.co/fair_day_v3_iphone_1.jpg"
                  },
                  mode: "FIT_XY",
                  "style": {
                    "size": {
                      "width": {
                        "value": 100,
                        "type": "PERCENT"
                      },
                      "height": {
                        "value": 100,
                        "type": "PERCENT"
                      }
                    }
                  }
                }
            }
   ]
}
```

That's why I need it public 

Thanks for understanding

**Comment by dantes-git** [comment]
When creating a component that contains another component (ServerDrivenComponent), it may be necessary to create a BeagleView. In the case of your example, create for each item in the _cards_ array.

`BeagleView(_component: ServerDrivenComponent)`

This is required if your component is built using autolayout and you need to configure the position of each card manually.

**Comment by dantes-git** [comment]
I understand now, we really can gain flexibility by exposing this api in this case. If this change solves the layout problem we can include it in the next release. Have you already tested your component by calling the BeagleScreenViewController init that accepts a component?

**Comment by iSallyAbdallah** [comment]
yes , and it's works great

**Comment by dantes-git** [comment]
Released in 2.1.0
